src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp

```diff
@@ -510,33 +514,52 @@ real64 CompositionalMultiphaseFVM::scalingForSystemSolution( DomainPartition & d
       {
         scalingFactor = std::min( scalingFactor, subRegionData.localMinVal );
       }
-      maxDeltaPres  = std::max( maxDeltaPres, subRegionData.localMaxDeltaPres );
-      maxDeltaCompDens = std::max( maxDeltaCompDens, subRegionData.localMaxDeltaCompDens );
-      maxDeltaTemp = std::max( maxDeltaTemp, subRegionData.localMaxDeltaTemp );
+
+      maxRegionDeltaPresLoc.push_back( {subRegionData.localMaxDeltaPres, localToGlobalMap[subRegionData.localMaxDeltaPresLoc]} );
       minPresScalingFactor = std::min( minPresScalingFactor, subRegionData.localMinPresScalingFactor );
+
+      maxRegionDeltaCompDensLoc.push_back( {subRegionData.localMaxDeltaCompDens, localToGlobalMap[subRegionData.localMaxDeltaCompDensLoc]} );
       minCompDensScalingFactor = std::min( minCompDensScalingFactor, subRegionData.localMinCompDensScalingFactor );
-      minTempScalingFactor = std::min( minTempScalingFactor, subRegionData.localMinTempScalingFactor );
+
+      if( m_isThermal )
+      {
+        maxRegionDeltaTempLoc.push_back( {subRegionData.localMaxDeltaTemp, localToGlobalMap[subRegionData.localMaxDeltaTempLoc]} );
+        minTempScalingFactor = std::min( minTempScalingFactor, subRegionData.localMinTempScalingFactor );
+      }
+
     } );
   } );
 
+  auto [localMaxDeltaPres, localMaxPresLoc] = *max_element( begin( maxRegionDeltaPresLoc ), end( maxRegionDeltaPresLoc ), []( auto & lhs, auto & rhs )   {
+    return std::get< 0 >( lhs ) < std::get< 0 >( rhs );
+  } );
+  auto globalMaxDeltaPres = MpiWrapper::tupleMaxLoc( std::tuple< real64, globalIndex >( localMaxDeltaPres, localMaxPresLoc ));
+
+  auto [ localMaxDeltaCompDens, localMaxCompDensLoc ] = *max_element( begin( maxRegionDeltaCompDensLoc ), end( maxRegionDeltaCompDensLoc ), []( auto & lhs, auto & rhs )   {
+    return std::get< 0 >( lhs ) < std::get< 0 >( rhs );
+  } );
+  auto globalMaxDeltaCompDens = MpiWrapper::tupleMaxLoc( std::tuple< real64, globalIndex >( localMaxDeltaCompDens, localMaxCompDensLoc ));
+
   scalingFactor = MpiWrapper::min( scalingFactor );
-  maxDeltaPres  = MpiWrapper::max( maxDeltaPres );
-  maxDeltaCompDens = MpiWrapper::max( maxDeltaCompDens );
   minPresScalingFactor = MpiWrapper::min( minPresScalingFactor );
   minCompDensScalingFactor = MpiWrapper::min( minCompDensScalingFactor );
 
   string const massUnit = m_useMass ? "kg/m3" : "mol/m3";
-  GEOS_LOG_LEVEL_RANK_0( 1, GEOS_FMT( "        {}: Max pressure change = {} Pa (before scaling)",
-                                      getName(), GEOS_FMT( "{:.{}f}", maxDeltaPres, 3 ) ) );
-  GEOS_LOG_LEVEL_RANK_0( 1, GEOS_FMT( "        {}: Max component density change = {} {} (before scaling)",
-                                      getName(), GEOS_FMT( "{:.{}f}", maxDeltaCompDens, 3 ), massUnit ) );
+  GEOS_LOG_LEVEL_RANK_0( 1, GEOS_FMT( "        {}: Max pressure change = {} Pa (before scaling) at global index {}",
```

Comment:
might be a bit confusing for a user but up to you - i would use "cell" instead of "global index"

Reply:
agreed